chrome/browser/android/adblock/adblock_bridge.cc

```diff
@@ -197,24 +190,22 @@ void handleOnLoad(content::WebContents* webContents, int frameTreeNodeId) {
         filterEngine->IsElemhideWhitelisted(url, referrers)) {
        LOG(WARNING) << "Adblock: element hiding - whitelisted";
     } else {
-      // generate JS
-      LOG(WARNING) << "Adblock: element hiding - generating JS ...";
       std::string domain = filterEngine->GetHostFromURL(url);
-      std::string js = generateJavascript(filterEngine, url, domain);
-      LOG(WARNING) << "Adblock: element hiding - generated JS";
 
       // run JS
       content::RenderFrameHost* frameHost = (frameTreeNodeId
         ? webContents->UnsafeFindFrameByFrameTreeNodeId(frameTreeNodeId)
         : webContents->GetMainFrame());
 
       if (frameHost) {
-        frameHost->ExecuteJavaScriptInIsolatedWorld(
-          base::UTF8ToUTF16(js),
-          content::RenderFrameHost::JavaScriptResultCallback(),
-          ISOLATED_WORLD_ID_ADBLOCK);
-
-        LOG(WARNING) << "Adblock: element hiding - called JS in frame"
+        LOG(WARNING) << "Adblock: element hiding - generating stylesheet ...";
+        std::string stylesheet = generateStylesheet(filterEngine, url, domain);
+        LOG(WARNING) << "Adblock: element hiding - generated stylesheet";
+        if (stylesheet.length() > 0) {
+          frameHost->InsertUserStylesheet(stylesheet);
```

Comment:
It seems a pretty notable change, could you please also include it in the merge request description?